Entity/Components/BuffComponent.gd

```diff
@@ -9,12 +9,19 @@ var current_buffs: Array[BuffBase]
 func add_buff(new_buff: BuffBase, buff_applier: Entity) -> void:
 	# duplicate the buff so we aren't modifying the base
 	var buff_copy: BuffBase = new_buff.duplicate()
-
-	current_buffs.append(buff_copy)
 	
-	assert(entity_owner != null, "BuffComponent has no owner. Please call init on Entity.")
+	# see if the buff was already applied. If so, add to the duration instead of applying
+	var buff_index: int = current_buffs.find(buff_copy)
```

Comment:
how does the `find` function behaves for self defined classes ?

Reply:
Actually that doesn't make sense now that I think about it, it wouldn't be able to find it in this case. Maybe it's searching by class type? It was adding to the duration from my testing, but maybe need to double check it works with lots of buffs in the array